akka-http-core/src/test/scala/akka/http/impl/engine/server/HttpServerSpec.scala

```diff
@@ -989,6 +989,28 @@ class HttpServerSpec extends AkkaSpec(
       shutdownBlueprint()
     })
 
+    "support remote-address-attribute" in assertAllStagesStopped(new TestSetup {
+      lazy val theAddress = InetAddress.getByName("127.5.2.1")
+
+      override def settings: ServerSettings = super.settings.withRemoteAddressAttribute(true)
+
+      override def modifyServer(server: ServerLayer): ServerLayer = {
+        BidiFlow.fromGraph(server.withAttributes(
+          HttpAttributes.remoteAddress(new InetSocketAddress(theAddress, 8080))
+        ))
+      }
+
+      send(
+        """GET / HTTP/1.1
+            |Host: example.com
+            |
+            |""".stripMarginWithNewline("\r\n"))
+
+      val request = expectRequest()
+      request.attributes(HttpRequest.AttributeKeys.remoteAddress) should equal(RemoteAddress(theAddress, Some(8080)))
```

Comment:
It might be nice to add those other headers with other addresses, to show that these headers don't change the attribute

Reply:
Good idea!